runtime-sdk/src/modules/core/types.rs

```diff
@@ -41,6 +45,17 @@ pub struct CallDataPublicKeyQueryResponse {
     pub epoch: u64,
 }
 
+/// Response to the public key query.
+#[derive(Clone, Debug, Default, cbor::Encode, cbor::Decode)]
+pub struct KeyManagerPublicKeyQueryResponse {
```

Comment:
Response to the public key query should probably only contain the public key. The request should contain the other two fields. So the name is a bit misleading (and long) and could be changed.

Reply:
I'm following the same naming scheme as `CallDataPublicKeyQueryResponse`.

It needs to return `key_pair_id` to be able to verify the signed ephemeral key, as that's not known ahead of time.

The `runtime_id` should be moved to `CallDataPublicKeyQueryResponse` as it's the runtime which we're encrypting the calldata for.